sksparse/cholmod.pyx

```diff
@@ -764,10 +947,10 @@ cdef class Factor:
         """
 
         return self(sparse.eye(self._factor.n, self._factor.n,
-                               dtype=_np_dtype_for(self._factor.xtype),
+                               dtype=_np_dtype_for_data(self._factor.xtype),
                                format="csc"))
 
-def analyze(A, mode="auto", ordering_method="default"):
+def analyze(A, mode="auto", ordering_method="default", use_long=None):
```

Comment:
The problem is that the input type is not the only criterion. It is possible, that the input matrix ``A`` uses 32-bit indices which might be reasonable if the number of non-zero entries and the shape is in the 32-bit integer range. But it might be that the cholesky factor ``L`` has much more non-zero entries so that 64-bit indices are needed here. This is not known apriori. If the user known that 32-bit indices are not enough, he/she can force the use of 64-bit indices with ``use_long=True``.